src/rpc/misc.cpp

```diff
@@ -138,6 +139,80 @@ static UniValue createmultisig(const JSONRPCRequest& request)
     return result;
 }
 
+UniValue deriveaddress(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.empty() || request.params.size() > 3)
+        throw std::runtime_error(
+            "deriveaddress \"descriptor\" ( \"start\" \"end\" )\n"
+            "\nDerives the address corresponding to an output descriptor.\n"
+            "Examples of output descriptors are:\n"
+            "    pkh(<pubkey>)                        P2PKH outputs for the given pubkey\n"
+            "    wpkh(<pubkey>)                       Native segwit P2PKH outputs for the given pubkey\n"
+            "    sh(multi(<n>,<pubkey>,<pubkey>,...)) P2SH-multisig outputs for the given threshold and pubkeys\n"
+            "    raw(<hex script>)                    Outputs whose scriptPubKey equals the specified hex scripts\n"
+            "\nIn the above, <pubkey> either refers to a fixed public key in hexadecimal notation, or to an xpub/xprv optionally followed by one\n"
+            "or more path elements separated by \"/\", where \"h\" represents a hardened child key.\n"
+            "For more information on output descriptors, see the documentation in the doc/descriptors.md file.\n"
+            "\nArguments:\n"
+            "1. \"descriptor\"                     (string, required) The descriptor\n"
+            "2. \"n\" / \"start\"                    (integer, optional) If a ranged descriptor is used, this specifies the number of keys, or start of the range to import.\n"
+            "3. \"end\"                            (integer, optional) If a ranged descriptor is used, this specifies the end of the range to import.\n"
+            "\nResult:\n"
+            "[ address ] (array) the derived addresses\n"
+            "\nExamples:\n"
+            "\nFirst three native segwit receive addresses\n"
+            + HelpExampleCli("deriveaddress", "\"wpkh([d34db33f/84h/0h/0h]xpub6DJ2dNUysrn5Vt36jH2KLBT2i1auw1tTSSomg8PhqNiUtx8QX2SvC9nrHu81fT41fvDUnhMjEzQgXnQjKEu3oaqMSzhSrHMxyyoEAmUHQbY/0/*)\" 3") +
+            "\nFirst two P2SH-segwit addresses\n"
+            + HelpExampleCli("deriveaddress", "\"sh(wpkh([d34db33f/49h/0h/0h]xpub6DJ2dNUysrn5Vt36jH2KLBT2i1auw1tTSSomg8PhqNiUtx8QX2SvC9nrHu81fT41fvDUnhMjEzQgXnQjKEu3oaqMSzhSrHMxyyoEAmUHQbY/0/*))\" 0 1") +
+            ""
+        );
+
+    RPCTypeCheck(request.params, {UniValue::VSTR, UniValue::VNUM, UniValue::VNUM});
+    const std::string desc_str = request.params[0].get_str();
+
+    int range_start = 0;
+    int range_end = 0;
+
+    if (request.params.size() >= 2) {
+        if (request.params.size() >= 3) {
+            range_start = request.params[1].get_int();
+            range_end = request.params[2].get_int();
+        } else {
+            range_end = request.params[1].get_int() - 1;
```

Comment:
I feel like this is unnecessarily complicated, I'd prefer just start and end parameters without the "n" parameter